src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -221,6 +226,40 @@ private void PerformSecurityChecks()
             {
                 HasSuspiciousContent = true;
             }
+
+            // We can skip code scan if the ScriptBlockAst is from a .psd1 file and it's in fact a safe HashtableAst.
+            bool CanSkipCodeScan()
```

Comment:
The comment and code do not match - it is possible to call this method where the filename is not a psd1 file.

This code is also not future proof - it would be easy to miss the need for adding code here when the ast is enhanced. I don't have a simple suggestion on how to fix though.

What you really want is structural equivalence to an oracle, ignoring the children of the hashtable, then calling `IsAstSafe`, but I can't think of a fast way to implement that.

Reply:
Updated the local function name and comment to make them match.

I agree the code is not future proof ... I cannot think of a better solution. I added a `NOTE` comment in the local function to call it out.